iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -340,18 +340,30 @@ static LogicalResult setRootConfig(
       workloadPerWorkgroup,
       /*workgroupSize =*/ArrayRef<int64_t>{});
 
+  SmallVector<int64_t> matmulL1TileSizes(clMatmulL1TileSize.begin(),
+                                         clMatmulL1TileSize.end());
+  if (clMatmulL1TileSize.empty()) {
+    Optional<std::string> triple = getTargetTriple(entryPointFn);
+    if (triple && triple.getValue().find("x86_64") != std::string::npos) {
+      matmulL1TileSizes.append(3 + isBatchMatmul, 16);
```

Comment:
This will also tile the batch dimension (if exists) with `16`? Is that intended?

Reply:
Havent looked into the details here, but lets not tile the batch dimension. If needed add a lit test to make sure we know what happening here.

Reply:
The batch dim is actually not tiled in the below lines. Anyway, it's dropped and fall back to previous approach -- using same tile size for all tile sizes.